### ISSUE TITLE
`no-floating-promises` with `allowForKnownSafeCalls` not working

### ISSUE THREAD

**Issue by theoludwig** [oxc-project/tsgolint#135]
Hey!

By reading https://oxc.rs/blog/2025-08-17-oxlint-type-aware, I wanted to try oxlint with type aware. And found out that for the following configuration:
```json
{
  "@typescript-eslint/no-floating-promises": [
    "error",
    {
      "allowForKnownSafeCalls": [
        {
          "from": "package",
          "name": ["test", "it", "suite", "describe"],
          "package": "node:test"
        }
      ]
    }
  ],
}
```

`allowForKnownSafeCalls` seems to be ignored in oxlint, for reference, TypeScript ESLint documentation: https://typescript-eslint.io/rules/no-floating-promises/#allowforknownsafecalls

Running `oxlint --type-aware` reports more errors than `eslint` in case it shouldn't, notably `node:test` functions like `it` and `describe` don't need to be awaited.

**Comment by theoludwig** [comment]
It seems that it is already handled and tested. :thinking:
https://github.com/oxc-project/tsgolint/blob/894e03dc298f06927d6daaaba72e98646c85288f/internal/rules/no_floating_promises/no_floating_promises_test.go#L779-L788

**Comment by theoludwig** [comment]
Minimal reproduction example: https://github.com/theoludwig/oxlint-tsgolint-issue-135

**Comment by kvnxiao** [comment]
I think this applies to all rules with custom option configs, those custom configs don't seem to be respected.

**Comment by camc314** [comment]
Configuring rules is not supported during this initial preview, it will be coming soon.

**Comment by kvnxiao** [comment]
That makes sense @camc314 ! I assume this also applies `// oxlint-disable` comments inline in the code?

**Comment by camchenry** [comment]
This rule is supported now and can be configured, but sounds like it still has a bug in it: https://github.com/oxc-project/tsgolint/issues/51#issuecomment-3542022793. My hunch is that maybe passing a string array for the name may not work yet, I don't think we have any tests for that currently.

**Comment by camc314** [comment]
Should be fixed now

**Comment by camc314** [comment]
I just kicked off `0.7.2` release

**Comment by theoludwig** [comment]
> Should be fixed now

Not yet apparently. 😅 
I updated the reproduction example to use oxlint v1.29.0 and tsgoling v0.7.2, and the issue still appears.
https://github.com/theoludwig/oxlint-tsgolint-issue-135/commit/9dd7e3fb39ff9dc60269d0ecf4b457c0ffa566b8

```sh
➜ node --run lint       

  × typescript-eslint(no-floating-promises): Promises must be awaited.
    ╭─[src/add.test.ts:5:1]
  4 │     
  5 │ ╭─▶ describe("add", () => {
  6 │ │     it("should return the sum of two numbers", () => {
  7 │ │       // Arrange - Given
  8 │ │       const a = 5
  9 │ │       const b = 3
 10 │ │   
 11 │ │       // Act - When
 12 │ │       const actual = add(a, b)
 13 │ │   
 14 │ │       // Assert - Then
 15 │ │       const expected = 8
 16 │ │       assert.strictEqual(actual, expected)
 17 │ │     })
 18 │ ╰─▶ })
    ╰────
  help: The promise must end with a call to .catch, or end with a call to .then with a rejection handler, or be explicitly marked as ignored with the `void` operator.
```

**Comment by camc314** [comment]
I think i mucked up a rebase somewhere

**Comment by camc314** [comment]
Yikes this was a pain to narrow down

**Comment by camc314** [comment]
3rd times a charm hopefully. 0.7.3 is releasing now 🤞

**Comment by theoludwig** [comment]
Awesome! Thank you for making oxlint backwards compatible by keeping what is great about ESLint and TypeScript ESLint (but also of course ditching what isn't #425).
